### PR TITLE
feat: added support for proxy agent in https call

### PR DESCRIPTION
**Pull request by sinha-sahil**
- added parameter for passing https/http agent in https call made for getAccessToken
- updated types

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -75,6 +76,7 @@ class ShopifyToken {
     this.sharedSecret = options.sharedSecret;
     this.redirectUri = options.redirectUri;
     this.apiKey = options.apiKey;
+    this.proxyAgent = 'proxyAgent' in options ? options.proxyAgent : null;
```
Comment (lpinca) [inline index.js:79]: It can actually be any agent so I would rename the option to just `agent`.

**Comment by lpinca** [comment]
The new option should also be added to the README.

File: index.js (hunk 1 above)
Comment (sinha-sahil) [reply index.js]: Updated name to `agent`

File: index.js [hunk 2]
```diff
@@ -58,6 +58,7 @@ class ShopifyToken {
    * @param {String} options.apiKey The API Key for the app
    * @param {String} [options.accessMode] The API access mode
    * @param {Number} [options.timeout] The request timeout
+   * @param {https.Agent|http.Agent} [options.agent] The agent for all http/https calls
```
Comment (lpinca) [inline index.js:61]: The endpoint is HTTPS only so I think this should be only `https.Agent`, no?

File: README.md [hunk 3]
```diff
@@ -45,6 +45,9 @@ Creates a new `ShopifyToken` instance.
   token that respects the user's permission level when making API requests
   (called online access). This is strongly recommended for embedded apps.
   Defaults to offline access mode.
+- `agent` - Optional - An http/https agent which will be passed on to https
+  request made for obtaining auth token. This is useful when trying to obtain
+  token from a server which has constraints on internet access.
```
Comment (lpinca) [inline README.md:50]:
```suggestion
- `agent` - Optional - An HTTPS agent which will be passed to the HTTPS
  request made for obtaining the auth token. This is useful when trying to
  obtain a token from a server that has restrictions on internet access.
```

File: index.js (hunk 2 above)
Comment (sinha-sahil) [reply index.js:61]: Actually there can be different kind of tunnelling using agents. 
All possible combinations are:
1. Http-Over-Http
2. Http-Over-Https
3. Https-Over-Http
4. Https-Over-Https
In `getAccessToken` 's use case point 2 & point 4 are valid & useful. 
Hence added the support for `http.Agent` & `https.Agent`

File: index.js [hunk 4]
```diff
@@ -75,6 +76,7 @@ class ShopifyToken {
     this.sharedSecret = options.sharedSecret;
     this.redirectUri = options.redirectUri;
     this.apiKey = options.apiKey;
+    this.agent = 'agent' in options ? options.agent : null;
```
[the remark this answers is not in the document]
Comment (sinha-sahil) [reply index.js]: Updated the PR.

File: README.md (hunk 3 above)
Comment (sinha-sahil) [reply README.md]: Updated the README.md

File: index.js (hunk 2 above)
Comment (lpinca) [reply index.js:61]: The proxy can be HTTP or HTTPS but the endpoint is always HTTPs so I think it's 3 & 4 in your list. For example see https://github.com/delvedor/hpagent/tree/v1.2.0#usage and https://github.com/delvedor/hpagent/blob/v1.2.0/index.js#L64.

**Review by lpinca — APPROVED** [review]

**Comment by lpinca** [comment]
Thank you.

**Comment by sinha-sahil** [comment]
@lpinca Please publish this change to the NPM as well. I need it one of my applications 😅 (currently pointing it to git repo link, will move it back to NPM package version once this is published).

**Pull request merged**